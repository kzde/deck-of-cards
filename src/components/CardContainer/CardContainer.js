import React from 'react';
import Card from '../../components/Card/Card';
import './CardContainer.scss';

const CardContainer = (props) => {

    return (
        <div className="CardContainer">
            {props.cards.map((card, index ) => {
                return (<Card key={index} card={card} />)
            })}
        </div>
    )
}

export default CardContainer;