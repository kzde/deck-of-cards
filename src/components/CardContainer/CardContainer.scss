.CardContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;

    .onecarddisplay {
        display: flex;
        flex-direction: column;
        align-items: center;
        margin: 0 auto;
        margin-top: 30;

        div {
            margin: 15px
        }
    }
}