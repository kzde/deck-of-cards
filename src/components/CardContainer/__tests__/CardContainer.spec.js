import React from 'react';
import renderer from 'react-test-renderer';
import CardContainer from '../CardContainer';
jest.mock('../../Card/Card', () => 'Card');


const props = {
    cards: [{
        group: 'club',
        value: 'queen'
    },{
        group: 'spade',
        value: 'ase'
    }]
}

describe('Card', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<CardContainer {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})