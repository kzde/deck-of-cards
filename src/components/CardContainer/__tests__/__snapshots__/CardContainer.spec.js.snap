// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should renders correctly 1`] = `
<div
  className="CardContainer"
>
  <Card
    card={
      Object {
        "group": "club",
        "value": "queen",
      }
    }
  />
  <Card
    card={
      Object {
        "group": "spade",
        "value": "ase",
      }
    }
  />
</div>
`;
